### PR TITLE
Delete class and delete property functionality

### PR DESCRIPTION
**Pull request by IvanJK97**
Added delete class from schema and delete property from class functionality in ecschema-metadata and ecschema-editing. Also wrote corresponding tests. Added a few more property tests of different primitive types.

### CODE REVIEW

File: core/ecschema-editing/src/Editing/CustomAttributes.ts [hunk 1]
```diff
@@ -43,6 +43,17 @@ export class CustomAttributes extends ECClasses {
     return { itemKey: newClass.key };
   }
 
+  public async delete(schemaKey: SchemaKey, name: string): Promise<SchemaItemEditResults> {
+    const schema = await this._schemaEditor.getSchema(schemaKey);
```
Comment (pmconne) [inline core/ecschema-editing/src/Editing/CustomAttributes.ts:47]: I count 5 identical-looking copies of this function...possible to consolidate them?

**Review by pmconne — CHANGES_REQUESTED** [review]

File: core/ecschema-editing/src/Editing/CustomAttributes.ts (hunk 1 above)
Comment (IvanJK97) [reply core/ecschema-editing/src/Editing/CustomAttributes.ts]: I moved delete function to ECClasses so there is only one copy of this.

**Review by pmconne — APPROVED** [review]

**Review by ColinKerr — CHANGES_REQUESTED** [review]
This PR adds tslint rules, it surely should not

File: core/ecschema-metadata/src/test/Metadata/Class.test.ts [hunk 2]
```diff
@@ -99,6 +100,181 @@ describe("ECClass", () => {
...
+      expect(entityClass.properties).to.be.undefined;
+      expect(await entityClass.getProperty("TestProp", true)).to.be.undefined;
+
+      (entityClass as ECClass as MutableClass).deletePropertySync("TestProp");
+
+      expect(entityClass.properties).to.be.undefined;
+      expect(await entityClass.getProperty("TestProp", true)).to.be.undefined;
+    });
+
+    it("should do nothing if a property is already deleted", async () => {
```
Comment (IvanJK97) [inline core/ecschema-metadata/src/test/Metadata/Class.test.ts:131]: Think this test and below's `deleteProperty` and `deletePropertySync` is flipped.

**Review by christophermlawson — APPROVED** [review]

File: core/ecschema-editing/src/test/Editing/Mixins.test.ts [hunk 3]
```diff
@@ -26,4 +26,29 @@ describe("Mixins tests", () => {
...
   });
+
+  it("should delete a mixin", async () => {
+    const schema = await testEditor.schemaContext.getCachedSchema(testKey);
+    const className = "testMixin";
+    const mixinResult = await testEditor.mixins.create(testKey, className, entityKey);
+    const mixin = await schema?.getItem(className);
+    expect(mixin).not.undefined;
+
+    /* eslint-disable @typescript-eslint/no-non-null-asserted-optional-chain */
```
Comment (ColinKerr) [inline core/ecschema-editing/src/test/Editing/Mixins.test.ts:37]: why is this needed?  Why couldn't we right the code so it does not violate this lint rule?  Same for the the next case where this rule is ignored.

Comment (christophermlawson) [reply core/ecschema-editing/src/test/Editing/Mixins.test.ts]: Done

**Review by christophermlawson — APPROVED** [review]

**Review by ColinKerr — APPROVED** [review]

**Review by ColinKerr — APPROVED** [review]

**Review by calebmshafer — APPROVED** [review]
